src/gridfs/download.rs

```diff
@@ -0,0 +1,181 @@
+use tokio::io::AsyncWriteExt;
+use tokio_util::compat::FuturesAsyncWriteCompatExt;
+
+use super::{options::GridFsDownloadByNameOptions, FilesCollectionDocument, GridFsBucket};
+use crate::{
+    bson::{doc, Bson},
+    error::{ErrorKind, Result},
+    options::{FindOneOptions, FindOptions},
+};
+
+impl GridFsBucket {
```

Comment:
Could we provide doctest examples for each of these? (sorry i know it will be a lot of similar examples, but I think it's something users really appreciate)